test/integration/gardenlet/backupbucket/backupbucket_suite_test.go

```diff
@@ -0,0 +1,265 @@
+	testNamespace = &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			// create dedicated namespace for each test run, so that we can run multiple tests concurrently for stress tests
+			GenerateName: "garden-",
+		},
+	}
+	Expect(testClient.Create(ctx, testNamespace)).To(Succeed())
+	log.Info("Created project Namespace for test", "namespaceName", testNamespace.Name)
+
+	DeferCleanup(func() {
+		By("deleting project namespace")
+		Expect(testClient.Delete(ctx, testNamespace)).To(Or(Succeed(), BeNotFoundError()))
+	})
+
+	By("creating seed")
+	seed = &gardencorev1beta1.Seed{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "seed-",
+			Labels:       map[string]string{testID: testRunID},
+		},
+		Spec: gardencorev1beta1.SeedSpec{
+			Provider: gardencorev1beta1.SeedProvider{
+				Region: "region",
+				Type:   "providerType",
+			},
+			Networks: gardencorev1beta1.SeedNetworks{
+				Pods:     "10.0.0.0/16",
+				Services: "10.1.0.0/16",
+				Nodes:    pointer.String("10.2.0.0/16"),
+			},
+			DNS: gardencorev1beta1.SeedDNS{
+				IngressDomain: pointer.String("someingress.example.com"),
+			},
+		},
+	}
+	Expect(testClient.Create(ctx, seed)).To(Succeed())
+	log.Info("Created Seed for test", "seed", seed.Name)
+
+	DeferCleanup(func() {
+		By("deleting seed")
+		Expect(testClient.Delete(ctx, seed)).To(Or(Succeed(), BeNotFoundError()))
+	})
+
+	By("creating garden namespace")
+	gardenNamespace = &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "garden",
+		},
+	}
+
+	Expect(testClient.Create(ctx, gardenNamespace)).To(Or(Succeed(), BeAlreadyExistsError()))
+	log.Info("Created namespace for test", "namespaceName", gardenNamespace)
+
+	DeferCleanup(func() {
+		By("deleting garden namespace")
+		Expect(testClient.Delete(ctx, gardenNamespace)).To(Or(Succeed(), BeNotFoundError()))
+	})
+
+	// Both the garden and seed cluster are same in this environment.
+	// So we create this to differentiate between the two garden namespaces.
+	By("creating seed garden namespace")
+	seedGardenNamespace = &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "seed-garden-",
+		},
+	}
+
+	Expect(testClient.Create(ctx, seedGardenNamespace)).To(Succeed())
+	log.Info("Created namespace for test", "namespaceName", gardenNamespace)
+
+	DeferCleanup(func() {
+		By("deleting garden namespace")
+		Expect(testClient.Delete(ctx, seedGardenNamespace)).To(Or(Succeed(), BeNotFoundError()))
+	})
+
+	By("Create Secret")
+	secret = &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "test-secret-",
+			Namespace:    gardenNamespace.Name,
+			Labels:       map[string]string{testID: testRunID},
+		},
+		Data: map[string][]byte{
+			"foo": []byte("bar"),
+		},
+	}
+
+	Expect(testClient.Create(ctx, secret)).To(Succeed())
+	log.Info("Created secret for test", "secret", client.ObjectKeyFromObject(secret))
+
+	DeferCleanup(func() {
+		By("Delete Secret")
+		Expect(testClient.Delete(ctx, secret)).To(Succeed())
+	})
```

Comment:
Why do you create this here and not in the `BeforeEach` in the `_test.go`?

Reply:
Also, maybe call it `gardenSecret`? Generally, you might want to prefix the names of the resources with `garden` or `seed` so that it's clear what we are talking about.